mypy/semanal.py

```diff
@@ -2366,11 +2374,19 @@ def fail_typeddict_arg(self, message: str,
 
     def build_typeddict_typeinfo(self, name: str, items: List[str],
                                  types: List[Type]) -> TypeInfo:
-        mapping_value_type = join.join_type_list(types)
         fallback = (self.named_type_or_none('typing.Mapping',
```

Comment:
Yeah, we could special case additional methods, but it wouldn't still be safe: a TypedDict is a subtype of the fallback type, and if the fallback type is `Dict`, we could upcast a TypedDict to the `Dict`type  and do things that the TypedDict wouldn't support, such as setting items with incompatible value types, or deleting required keys.